system/system_monitor/src/net_monitor/net_monitor.cpp

```diff
@@ -492,5 +499,92 @@ void NetMonitor::monitorTraffic(diagnostic_updater::DiagnosticStatusWrapper & st
   SystemMonitorUtility::stopMeasurement(t_start, stat);
 }
 
+void NetMonitor::checkReassemblesFailed(diagnostic_updater::DiagnosticStatusWrapper & stat)
+{
+  // Remember start time to measure elapsed time
+  const auto t_start = SystemMonitorUtility::startMeasurement();
+
+  int whole_level = DiagStatus::OK;
+  std::string error_str;
+  uint64_t total_reassembles_failed = 0;
+  uint64_t unit_reassembles_failed = 0;
+
+  if (getReassemblesFailed(total_reassembles_failed)) {
+    reassembles_failed_queue_.push_back(total_reassembles_failed - last_reassembles_failed_);
+    while (reassembles_failed_queue_.size() > reassembles_failed_check_duration_) {
+      reassembles_failed_queue_.pop_front();
+    }
+
+    for (auto reassembles_failed : reassembles_failed_queue_) {
+      unit_reassembles_failed += reassembles_failed;
+    }
+
+    stat.add(fmt::format("total packet reassembles failed"), total_reassembles_failed);
+    stat.add(fmt::format("packet reassembles failed per unit time"), unit_reassembles_failed);
+
+    if (unit_reassembles_failed >= reassembles_failed_check_count_) {
+      whole_level = std::max(whole_level, static_cast<int>(DiagStatus::WARN));
+      error_str = "reassembles failed";
+    }
+
+    last_reassembles_failed_ = total_reassembles_failed;
+  } else {
+    reassembles_failed_queue_.push_back(0);
+    whole_level = std::max(whole_level, static_cast<int>(DiagStatus::ERROR));
+    error_str = "failed to read /proc/net/snmp";
+  }
+
+  if (!error_str.empty()) {
+    stat.summary(whole_level, error_str);
+  } else {
+    stat.summary(whole_level, "OK");
+  }
+
+  // Measure elapsed time since start time and report
+  SystemMonitorUtility::stopMeasurement(t_start, stat);
+}
+
+bool NetMonitor::getReassemblesFailed(uint64_t & reassembles_failed)
+{
+  std::ifstream ifs("/proc/net/snmp");
+  if (!ifs) {
+    return false;
+  }
+
+  std::string line;
+  while (std::getline(ifs, line)) {
+    std::vector<std::string> title_list;
+    boost::split(title_list, line, boost::is_space());
+
+    if (title_list.size() == 0) {
+      continue;
+    }
+
+    if (!std::getline(ifs, line)) {
+      return false;
+    }
+
+    std::vector<std::string> value_list;
+    boost::split(value_list, line, boost::is_space());
+
+    if (title_list[0] != "Ip:") {
```

Comment:
Maybe is this `value_list[0]`?

Reply:
@ito-san 
I replaced it with the code you provided.
2c33f22f1282476884173aa36baff81e710b9a79